text/0000-cargo-install.md

```diff
@@ -0,0 +1,271 @@
+possible.
+
+Put another way, enabling easily sharing code is one of Cargo's fundamental
+design goals, and expanding into binaries is simply an extension of Cargo's core
+functionality.
+
+# Detailed design
+
+The following new subcommand will be added to Cargo:
+
+```
+Install a crate onto the local system
+
+Installing new crates:
+    cargo install [options]
+    cargo install [options] [-p CRATE | --package CRATE] [--vers VERS]
+    cargo install [options] --git URL [--branch BRANCH | --tag TAG | --rev SHA]
+    cargo install [options] --path PATH
+
+Managing installed crates:
+    cargo install [options] --list
+    cargo install [options] --update [SPEC | --all]
+
+Options:
+    -h, --help              Print this message
+    -j N, --jobs N          The number of jobs to run in parallel
+    --features FEATURES     Space-separated list of features to activate
+    --no-default-features   Do not build the `default` feature
+    --debug                 Build in debug mode instead of release mode
+    --bin NAME              Only install the binary NAME
+    --example EXAMPLE       Install the example EXAMPLE instead of binaries
+    -p, --package CRATE     Install this crate from crates.io or select the
+                            package in a repository/path to install.
+    -v, --verbose           Use verbose output
+
+This command manages Cargo's local set of install binary crates. Only packages
+which have [[bin]] targets can be installed, and all binaries are installed into
+`$HOME/.cargo/bin` by default (or `$CARGO_HOME/bin` if you change the home
+directory).
+
+There are multiple methods of installing a new crate onto the system. The
+`cargo install` command with no arguments will install the current crate (as
+specifed by the current directory). Otherwise the `-p`, `--package`, `--git`,
+and `--path` options all specify the source from which a crate is being
+installed. The `-p` and `--package` options will download crates from crates.io.
+
+Crates from crates.io can optionally specify the version they wish to install
+via the `--vers` flags, and similarly packages from git repositories can
+optionally specify the branch, tag, or revision that should be installed. If a
+crate has multiple binaries, the `--bin` argument can selectively install only
+one of them, and if you'd rather install examples the `--example` argument can
+be used as well.
+
+The `--list` option will list all installed packages (and their versions). The
+`--update` option will update either the crate specified or all installed
+crates.
+```
+
+## Installing Crates
+
+Cargo attempts to be as flexible as possible in terms of installing crates from
+various locations and specifying what should be installed. All binaries will be
+stored in the **cargo-local** directory `CARGO_HOME/bin`. This is typically
+`$HOME/.cargo/bin` but the home directory can be modified via the `$CARGO_HOME`
+environment variable.
```

Comment:
Would binaries go into `$root` or `$root/bin`? With the latter, `cargo install` could later write things in `$root/share` or `$root/lib`, and there is precedent in `./configure` scripts to name it prefix rather than root.